js/interface.js

```diff
@@ -2610,11 +2696,47 @@ var DynamicLists = (function() {
           item.value = item.fieldValue;
         }
 
+        item.filterModifier = {
+          default: {
+            value: $('#date-field-' + item.id).val(),
+            offset: $('#number-field-' + item.id).val() || null,
+            useDeviceTimezone: $('#enable-timezone-default-' + item.id).prop('checked')
+          }
+        };
+
         if (item.logic === 'empty' || item.logic === 'notempty') {
-          item.valueType = null;
+          item.valueType = '';
           item.value = '';
         }
 
+        if (item.logic === 'datebetween') {
+          var valueDateFrom = $('#date-from-' + item.id).val();
+          var valueNumberFrom = $('#number-field-from-' + item.id).val();
+          var useDeviceTimezoneFrom = $('#enable-timezone-from-' + item.id).prop('checked');
+
+          var valueDateTo = $('#date-to-' + item.id).val();
+          var valueNumberTo = $('#number-field-to-' + item.id).val();
+          var useDeviceTimezoneTo = $('#enable-timezone-to-' + item.id).prop('checked');
+
+          item.filterModifier = {
+            default: {
```

Comment:
I assume the `datebewteen` logic requires modifiers for `from` and `to`. Why does it need values for the `default` modifier?

If it's not needed, we shouldn't be saving unnecessary data, which could increase the widget data and page size.